fedimint-cli/src/ng.rs

```diff
@@ -82,6 +97,83 @@ pub async fn handle_ng_command<D: IDatabase>(
 
             Ok(serde_json::to_value(amt).unwrap())
         }
+        ClientNg::LnPay { bolt11 } => {
+            let invoice = bolt11.clone();
+            let operation_id = bolt11.payment_hash().into_inner();
+            let ln_client = client
+                .get_module_client::<LightningClientModule>(LEGACY_HARDCODED_INSTANCE_ID_LN)
+                .unwrap();
+            let mut dbtx = client.db().begin_transaction().await;
+            let outgoing_ln_contract = ln_client
+                .create_outgoing_ln_contract(
+                    &mut dbtx.with_module_prefix(LEGACY_HARDCODED_INSTANCE_ID_LN),
+                    client.context(),
+                    invoice,
+                    rng,
+                )
+                .await?;
+
+            let (contract_id, amount) = match &outgoing_ln_contract {
+                LightningOutput::Contract(c) => {
+                    let contract_id = c.contract.contract_id();
+                    let amount = c.amount;
+                    (contract_id, amount)
+                }
+                LightningOutput::Offer(_) | LightningOutput::CancelOutgoing { .. } => {
+                    panic!()
+                } // FIXME: impl TryFrom
+            };
+
+            let mint_client = client
+                .get_module_client::<MintClientModule>(LEGACY_HARDCODED_INSTANCE_ID_MINT)
+                .unwrap();
+
+            let client_input = mint_client
+                .create_input(
+                    &mut dbtx.with_module_prefix(LEGACY_HARDCODED_INSTANCE_ID_MINT),
+                    operation_id,
+                    amount,
+                )
+                .await?;
+
+            let sm_gen = Box::new(move |funding_txid: TransactionId, _input_idx: u64| {
+                vec![LightningClientStateMachines::Pay(
+                    LightningPayStateMachine {
+                        common: LightningPayCommon {
+                            operation_id,
+                            federation_id: fed_id,
+                        },
+                        state: LightningPayStates::CreatedOutgoingLnContract(
+                            LightningPayCreatedOutgoingLnContract {
+                                funding_txid,
+                                contract_id,
+                            },
+                        ),
+                    },
+                )]
+            });
```

Comment:
This should be part of `create_outgoing_ln_contract`, the `ClientOutpput` struct already contains a state machine generator.